### DISCUSSION TITLE
possible to custom beforeunload?

### DISCUSSION THREAD

**Discussion by synasapmob** [community/community#118663] [Programming Help]
### Question

hello guys, possible to custom "confirm" before users leave page?
i researched so much but cannot found, or i missing something,
so i just wanna use [chakra-modal](https://v2.chakra-ui.com/docs/components/modal/usage) instead default

### Example
in each browser has a little the difference layout Ex:

**Window**
![Screen Shot 2024-04-12 at 10 10 35](https://github.com/community/community/assets/42969529/d5031220-711a-47f7-bb00-9b8d42bd4756)

**Brave**
![Screen Shot 2024-04-12 at 10 13 33](https://github.com/community/community/assets/42969529/41c35596-3636-42c8-b13f-378a7a4e7841)

**Edge**
![Screen Shot 2024-04-12 at 10 13 44](https://github.com/community/community/assets/42969529/7aea6a1d-0a3b-496f-8389-b65b8c771d5f)

Code:
```jsx
window.addEventListener('beforeunload',(event) => {

return event.preventDefault();
});
```

**Comment by ettaboyle** [comment]
Thanks for posting in the GitHub Community, @synasapmob!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. **The Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions).** 

This question should be in the [Programming Help category](https://github.com/orgs/community/discussions/categories/programming-help).  I've gone ahead and moved it for you. 

You are more likely to get a useful response if you are explicit about what your project entails, giving a few more details might help someone give you a nudge in the right direction.

Good luck!

**Reply by synasapmob** [reply]
thanks you, I'll updating details more

**Comment by 1112779648** [comment]
![IMG_8238](https://github.com/community/community/assets/166644182/0b04c4a0-39a5-4103-99e3-5405f3518df0)

**Reply by synasapmob** [reply]
sorry i can't understand

**Comment by croissain** [comment]
seem impossible

**Reply by synasapmob** [reply]
thanks you
